### ISSUE TITLE
C# - Tween target object has no property named: GlobalPosition.

### ISSUE THREAD

**Issue by Furfoxsnakes** [godotengine/godot#39858]
**Godot version:**

v3.2.2.stable.mono.official

**OS/device including version:**

Windows 10
GLES3

**Issue description:**

Error occurs when attempting to use "GlobalPosition" as property during InterpolateProperty method on Tween node. Error occurred after downloading Godot 3.2.2, although error now persists is a new 3.2.1 project as well

```
E 0:00:00.776   _build_interpolation: Tween target object has no property named: GlobalPosition.
  <C++ Error>   Condition "!prop_valid" is true. Returned: false
  <C++ Source>  scene/animation/tween.cpp:1308 @ _build_interpolation()
  <Stack Trace> :0 @ Boolean Godot.NativeCalls.godot_icall_8_866(IntPtr , IntPtr , IntPtr , IntPtr , System.Object , System.Object , System.Single& , Int32 , Int32 , System.Single& )()
                Tween.cs:411 @ Boolean Godot.Tween.InterpolateProperty(Godot.Object , Godot.NodePath , System.Object , System.Object , Single , Godot.Tween+TransitionType , Godot.Tween+EaseType , Single )()
                TweenTest.cs:12 @ void TweenTest._Ready()()
```

Expecting "GlobalPosition" to be a valid property when using mono/C#

Workaround involves using "global_position" instead during InterpolateProperty method.

**Steps to reproduce:**

- Create new project in Godot 3.2.2 or 3.2.1
- Add Sprite node as child (not required but there to visualize tween)
- Add Tween node as child
- Add C# script to Node2D node
- Use "this" as object and "GlobalPosition" as property on tween.InterpolateProperty method

**Minimal reproduction project:**

[Tween Test.zip](https://github.com/godotengine/godot/files/4839064/Tween.Test.zip)

**Comment by neikeq** [comment]
Right now, the Godot reflection API get/set/call doesn't support the PascalCase names the C# API uses. This is what tools like tween and the animation player use. You have to use the real names the engine uses, in this case: `global_position`. This is being followed in #16872.
Did this ever work before? I would be surprised if it did.

**Comment by Furfoxsnakes** [comment]
Yes, I can confirm that PascalCase was working before I updated to 3.2.2. I doubled checked my git history to make sure I wasn't mistaken, but it does show that I was using PascalCase.

Curiously it appears to be working in 3.2.1 again today. I created another project in 3.2.1 and was curious if the Rider IDE and it's Godot plugin was causing any issues, so I disabled external editor for Mono and used the internal editor. Everything ran properly using PascalCase. I then opened the project in 3.2.2 and property was invalid again. I then created a new project in 3.2.1 again and used Rider and found that PascalCase still worked.
